packages/core/core/telemetry/get-tracer.ts

```diff
@@ -0,0 +1,9 @@
+import opentelemetry from '@opentelemetry/api';
+
+// async to support dynamic imports / stubbing
+export async function getTracer() {
```

Comment:
Not a good idea to make it async. Tracing should equally support sync and async call tracing. If you want dynamic import - you should try to arrange it some other way.

Reply:
dynamic imports need to happen in async functions. it's not an issue because all AI SDK functions that we want to trace are async, unless i'm missing something?

Reply:
So, sync now?

Reply:
need to explore if there is any impact when using the functions in the browser (i.e. when certain node apis aren't available) - there are now e.g. chrome ai providers that do not need a backend

Reply:
changed to sync